gst/datarepo/gstdatareposink.c

```diff
@@ -293,6 +299,45 @@ gst_data_repo_sink_write_others (GstDataRepoSink * sink, GstBuffer * buffer)
   return GST_FLOW_OK;
 }
 
+/**
+ * @brief Function to validate flexible tensors
+ */
+static gboolean
+gst_data_repo_sink_validate_flexible_tensors (GstBuffer * buffer)
+{
+  guint num_tensors, i;
+  gsize hsize = 0;
+  GstMemory *in_mem[NNS_TENSOR_SIZE_LIMIT];
+  GstMapInfo in_info[NNS_TENSOR_SIZE_LIMIT];
+  GstTensorMetaInfo in_meta[NNS_TENSOR_SIZE_LIMIT];
```

Comment:
IMHO, with this code, the excess memory goes to waste.

Reply:
Thank you.
GstBuffer has the number of tensors(num_tensors) and we need to check if at least one is flexible.
The maximum value of num_tensors is NNS_TENSOR_SIZE_LIMIT. so we need to allow for wasting memory.

Reply:
Actually this function does not need all memory map info.
Also gst_tensor_meta_info_parse_memory() can validate meta.
I suggest simply using gst_tensor_meta_info_parse_memory().
e.g.
```
GstMemory *mem;
GstTensorMetaInfo meta;
gboolean ret;

for (;;) {
  mem = gst_tensor_buffer_get_nth_memory (buffer, i);
  ret = gst_tensor_meta_info_parse_memory (meta, mem);
  gst_memory_unref (mem);

  if (!ret)
    return FALSE;
}

return TRUE;
```

Reply:
wow, Thank you for your guide. :+1: